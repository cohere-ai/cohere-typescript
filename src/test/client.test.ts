import { describe, test } from "@jest/globals";
import { CohereClient } from "../index";

describe("v1 back compat", () => {
    test.each([
        "https://api.cohere.com/",
        "https://api.cohere.com/v1",
        "https://api.cohere.com/v1/",
        "https://api.cohere.com/v1//",
        "https://api.cohere.com/v2",
    ])("%s", async (environment) => {
        let url = ""

        const cohere = new CohereClient({
            token: "token",
            environment,
            fetcher: (async (opts) => {
                url = opts.url
                throw "we're done"
            })
        })

        try {
            await cohere.chat({ message: "hello" })
        } catch { }

        expect(url).toMatchSnapshot()
    })
});
