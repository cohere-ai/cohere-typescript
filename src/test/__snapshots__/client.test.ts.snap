// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v1 back compat https://api.cohere.com/ 1`] = `"https://api.cohere.com/v1/chat"`;

exports[`v1 back compat https://api.cohere.com/v1 1`] = `"https://api.cohere.com/v1/chat"`;

exports[`v1 back compat https://api.cohere.com/v1/ 1`] = `"https://api.cohere.com/v1/chat"`;

exports[`v1 back compat https://api.cohere.com/v1// 1`] = `"https://api.cohere.com/v1/v1/chat"`;

exports[`v1 back compat https://api.cohere.com/v2 1`] = `"https://api.cohere.com/v2/v1/chat"`;
